neurom/view/view.py

```diff
@@ -96,6 +84,27 @@ def get_default(variable, **kwargs):
     return kwargs.get(variable, default[variable])
 
 
+def _plane2col(plane):
+    '''take a string like 'xy', and return the indices from COLS.*'''
+    planes = ('xy', 'yx', 'xz', 'zx', 'yz', 'zy')
+    assert plane in planes, 'No such plane found! Please select one of: ' + str(planes)
+    return (getattr(COLS, plane[0].capitalize()),
+            getattr(COLS, plane[1].capitalize()), )
+
+
+def _get_linewidth(tr, parameters):
+    '''calculate the desired linewidth based on tree contents, and parameters'''
+    linewidth = get_default('linewidth', parameters)
+    # Definition of the linewidth according to diameter, if diameter is True.
+    if get_default('diameter', parameters):
+        # TODO: This was originally a numpy array. Did it have to be one?
+        scale = get_default('diameter_scale', parameters)
+        linewidth = [2 * segment_radius(s) * scale for s in iter_segments(tr)]
+        if not linewidth:
+            linewidth = get_default('linewidth', parameters)
+    return linewidth
+
+
 def tree(tr, plane='xy', new_fig=True, subplot=False, **kwargs):
```

Comment:
You mean not having **kwargs, right?  In principle I agree, but in this case it's hard, because this is a 'thin' wrapper over matplotlib, and there are a lot of options for plotting (the full list is here: https://github.com/BlueBrain/NeuroM/blob/master/neurom/view/common.py#L54) so then the function signature gets so long that it's hard to even read.

We've tried coming up with a work around before, but have never come to a good conclusion on how to do it (within the constraint of wanting single line viewing.)